addon/-private/system/store.js

```diff
@@ -1909,6 +1917,32 @@ Store = Service.extend({
       updated.
   */
   push(data) {
+    let pushed = this._push(data);
+
+    if (Array.isArray(pushed)) {
+      let records = pushed.map(function(internalModel) {
+        return internalModel.getRecord();
+      });
+      return records;
+    }
+
+    if (pushed === null) {
+      return null;
+    }
+
+    var record = pushed.getRecord();
+    return record;
+  },
+
+  /**
+    Push some data into the store, without creating materialized records.
+
+    @method _push
+    @private
+    @param {Object} data
+    @return {DS.InternalModel|Array} pushed InternalModel(s)
+   */
+  _push(data) {
```

Comment:
Struggling to find a better name here, any suggestions? `_shallowPush` or `_lightweightPush`? Because `_pushInternalModel` doesn't sound right to me...